### PR TITLE
Fix some bugs

### LINKED ISSUE #940: 建议更新功能data.name2uuid
### 您的建议是否与现存的某个问题相关?请描述问题?

_No response_

### 您认为还缺少什么?如何解决您的问题?

data.name2uuid
使用玩家名称获取uuid

### PR DESCRIPTION
**Pull request by KobeBryant114514**
## Description

LiteLoader
1. 修复 Player::setPlayerNbt 不能写入在线玩家nbt的问题
2. Player::getPlayerNbt 在获取不到玩家数据时，不再返回空的 CompoundTag
3. 新增 Player::getAllUuid
4. 新增 PlayerInfo::getUUIDfromXuid  #940 

ScriptEngine
1. mc.getPlayerNbt 在获取不到玩家数据时返回null，不再返回空的nbt对象
2. 新增data.name2uuid / data.xuid2uuid  #940 

在本地Build并测试以上更改，未发现问题

## Issue Fixed by this PR
Close #940 

- [x] Bug fix
- [x] New feature 
- [x] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://docs.litebds.com/#/zh_CN/Maintenance/README)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

File: ScriptEngine/src/api/DataAPI.cpp [hunk 1]
```diff
@@ -712,7 +714,7 @@ Local<Value> DataClass::xuid2name(const Arguments& args) {
     try {
         return String::newString(PlayerInfo::fromXuid(args[0].toStr()));
     }
-    CATCH("Fail in Xuid2Name!");
+    CATCH("Fail in XuidtoName!");
```
Comment (Jaciezyt) [inline ScriptEngine/src/api/DataAPI.cpp]:
```suggestion
    CATCH("Fail in XuidToName!");
```
下同

File: LiteLoader/src/llapi/mc/PlayerAPI.cpp [hunk 2]
```diff
@@ -1021,48 +1021,71 @@ bool Player::deletePlayerNbt(mce::UUID const& uuid) {
...
+        if (serverId.empty()) {
             return false;
         }
         bool res = true;
-        if(auto pl = Global<Level>->getPlayer(uuid)) {
+        if (auto pl = Global<Level>->getPlayer(uuid)) {
             auto playerTag = pl->getNbt();
-            for(int i = 0; i <= tags.size()-1; i++) {
-                if(data.get(tags[i]) == nullptr) {
+            for (int i = 0; i <= tags.size()-1; i++) {
```
Comment (Jaciezyt) [inline LiteLoader/src/llapi/mc/PlayerAPI.cpp:1087]: 格式还是有问题，建议clang-format一下
此处遍历vector建议使用for (auto& it : tags)

File: LiteLoader/src/llapi/mc/PlayerAPI.cpp [hunk 3]
```diff
@@ -1021,48 +1021,71 @@ bool Player::deletePlayerNbt(mce::UUID const& uuid) {
...
+        else {
+            return setOfflineNbt(uuid, nbt);
+        }
+    }
+    catch (const std::exception& exc) {
+        logger.error("Fail to set player nbt!\n{}",exc.what());
+    }
+}
+
+bool Player::setPlayerNbtTags(mce::UUID const& uuid, CompoundTag* nbt, vector<string> tags) {
```
Comment (Jaciezyt) [inline LiteLoader/src/llapi/mc/PlayerAPI.cpp]:
```suggestion
bool Player::setPlayerNbtTags(mce::UUID const& uuid, CompoundTag* nbt, const vector<string>& tags) {
```
若函数内没有对tags进行赋值，推荐使用常量左值

File: LiteLoader/include/llapi/PlayerInfoAPI.h [hunk 4]
```diff
@@ -60,6 +60,7 @@ LIAPI std::vector<Info> getAllPlayerInfo();
 LIAPI bool insert(std::string name, std::string xuid, std::string uuid);
 LIAPI std::string getXuid(std::string name);
 LIAPI std::string getUUID(std::string name);
+LIAPI std::string getUUIDfromXuid(xuid_t xuid) ;
```
Comment (Jaciezyt) [inline LiteLoader/include/llapi/PlayerInfoAPI.h]:
```suggestion
LIAPI std::string getUUIDByXuid(xuid_t xuid) ;
```

File: LiteLoader/src/llapi/mc/PlayerAPI.cpp [hunk 5]
```diff
@@ -1074,18 +1097,18 @@ bool Player::setPlayerNbtTags(mce::UUID const& uuid, CompoundTag& data, vector<s
             data.destroy();
             return res;
         }
-        else{
+        else {
             auto oridata = getOfflineNbt(uuid);
             CompoundTag& olddata = *oridata;
-            for(int i = 0; i <= tags.size()-1; i++) {
-                if(data.get(tags[i]) == nullptr) {
+            for (int i = 0; i <= tags.size()-1; i++) {
```
Comment (Jaciezyt) [inline LiteLoader/src/llapi/mc/PlayerAPI.cpp:1103]: 同上

**Review by Jaciezyt — CHANGES_REQUESTED** [review]

**Review by ShrBox — APPROVED** [review]

**Pull request merged**